decavision/dataset_preparation/generate_pseudolabels.py

```diff
@@ -0,0 +1,220 @@
+
+            Returns:
+                pseudo_data_path: A folder with both labeled and pseudo labeled images.
+        """
+
+        # Copy the training/labeled data to the destination folder where
+        # we will also store the pseudo labels.
+        copy_tree(self.train_data_path, self.pseudo_data_path)
+
+        # Make dictionary for classes and their index
+        class_names = sorted(os.listdir(self.train_data_path))
+        class_dict = {}
+        index = 0
+        for cat in class_names:
+            class_dict[cat] = index
+            index += 1
+
+        print("Generating pseudo labels...")
+        # Generate pseudo labels
+        unlabeled_image_paths = os.listdir(self.unlabeled_path)
+        print("There are {} unlabeled images.".format(
+            len(unlabeled_image_paths)))
+        raw_predictions_paths = []
+        raw_predictions = []  # single confidence value of predicted class
+        predicted_class = []  # predicted class index
+        raw_predictions_all = []  # confidences for all classes
+        for path in tqdm(unlabeled_image_paths[:100]):
+            # Load the image
+            img = np.array(self._load_img(os.path.join(self.unlabeled_path, path),
+                                          target_size=(299, 299)))
+            # Make predictions using trained model
+            y_pred = self.model.predict(np.expand_dims(img, axis=0), verbose=0)
+            # Get class index
+            y = np.argmax(y_pred)
+            # Get probability score
+            proba = y_pred[0][y]
+
+            predicted_class.append(y)
+            raw_predictions.append(proba)
+            raw_predictions_all.append(list(y_pred[0]))
+            raw_predictions_paths.append(path)
+
+        if not os.path.exists("outputs"):
+            os.mkdir("outputs")
+        if save_confidences_csv:
+            # 'Pseudo Class Names': pseudo_class_names,
+            print("Saving CSV with pseudo predictions.")
+            data = {'Filepaths': raw_predictions_paths,
+                    'Predicted Class': predicted_class,
+                    'Highest Confidence': raw_predictions,
+                    'All Class Predictions': raw_predictions_all}
+            df = pd.DataFrame(data)
+            df.to_csv("outputs/data.csv", index=False)
+
+        # Save pseudo labeled images
+        self._save_pseudo_labels(csv_path="outputs/data.csv",
+                                 threshold=threshold,
+                                 dictionary=class_dict)
+
+        if plot_confidences:
+            # Plot only if there are any CSV files.
+            if save_confidences_csv:
+                print("Plotting data.")
+                self._plot_confidence_scores(
+                    csv_path="outputs/data.csv", classes=class_names)
+            else:
+                print("No CSV file is present to plot the data.")
+
+
+if __name__ == '__main__':
```

Comment:
This is not necessary. The code in this script is made to be imported and not run from the console.

Reply:
Done!